pdfinfo: pdfinfo_to_buffer takes a password

PDF Info read a password-protected PDF without authenticating. Goes with
0005-pdfinfo-to-buffer.
---
diff --git a/source/tools/pdfinfo.c b/source/tools/pdfinfo.c
index 8c4f7dcc6..149c17c1b 100644
--- a/source/tools/pdfinfo.c
+++ b/source/tools/pdfinfo.c
@@ -1042,7 +1042,7 @@ pdfinfo_info(fz_context *ctx, fz_output *out, char *filename, char *password, in
 		fz_rethrow(ctx);
 }
 
-fz_buffer *pdfinfo_to_buffer(fz_context *ctx, const char *filename)
+fz_buffer *pdfinfo_to_buffer(fz_context *ctx, const char *filename, const char *password)
 {
 	fz_buffer *buf = NULL;
 	fz_output *out = NULL;
@@ -1057,6 +1057,9 @@ fz_buffer *pdfinfo_to_buffer(fz_context *ctx, const char *filename)
 	fz_try(ctx)
 	{
 		glo.doc = pdf_open_document(ctx, filename);
+		if (pdf_needs_password(ctx, glo.doc))
+			if (!pdf_authenticate_password(ctx, glo.doc, password))
+				fz_throw(ctx, FZ_ERROR_ARGUMENT, "cannot authenticate password: %s", filename);
 		glo.pagecount = pdf_count_pages(ctx, glo.doc);
 		showglobalinfo(ctx, &glo);
 		showinfo(ctx, &glo, (char *)filename, ALL, "1-N");
